### DISCUSSION TITLE
How to make custom filter

### DISCUSSION THREAD

**Discussion by abitme** [rappasoft/laravel-livewire-tables#1282] [Q&A]
Since i have many data to filter i want to use combobox headless ui as select filter so the options only show when user typing it. 
Is there a way how I can achieve this?
I'm currently use configurable areas but I don't know how to apply the filter after selecting the option

**Comment by lrljoe** [comment]
I created an addon package which is a bit out of date now as I'm busy working on Version 3
[https://github.com/LowerRockLabs/LaravelLivewireTablesAdvancedFilters](https://github.com/LowerRockLabs/LaravelLivewireTablesAdvancedFilters) but it should give you an idea of how you can implement it.

Version 3 is not too far away either, and there's some more filter options present in v3, as well as support for Livewire v3

**Reply by lrljoe** [reply]
If you get stuck, jump on the Discord and I'll help when I can :)

**Reply by abitme** [reply]
thanks for the answer, i'm gonna try it

**Reply by lrljoe** [reply]
Essentially you just need to make a basic livewire component that extends either the base filter, or an existing one.  Create a view, and include it in your table component.
It just needs to have the correct methods to work properly.
Take a look at the code in both the component itself, and the blades for the text filter, and multi-select dropdown filters, those will get you pretty close.
It'll be easier to do that than to try and do anything else to be honest!

The SlimSelect filter in the other package is probably the closest to what you'd want, but I did make it more granular than I intended to in terms of blades.

**Comment by lrljoe** [comment]
I'm currently working on two items for v3 for this

a CustomFilter that you just pass a blade to
a custom Livewire Component filter
